packaging/linux/build_and_push_packages.sh

```diff
@@ -55,96 +55,64 @@ version="$(cat "$build_dir/VERSION")"
-  for blob in $dot_deb_blobs ; do
-    s3cmd modify --remove-header "Cache-Control" "$blob"
-    s3cmd cp "$blob" "s3://$BUCKET_NAME/linux_binaries/deb/"
-  done
-  echo Unsetting .rpm Cache-Control headers...
-  dot_rpm_blobs="$(s3cmd ls -r "s3://$BUCKET_NAME/rpm" | awk '{print $4}' | grep '\.rpm$')"
-  for blob in $dot_rpm_blobs ; do
-    s3cmd modify --remove-header "Cache-Control" "$blob"
-    s3cmd cp "$blob" "s3://$BUCKET_NAME/linux_binaries/rpm/"
-  done
-
-  # Make yet another copy of the .deb and .rpm packages we just made, in a
-  # constant location for the friend-of-keybase instructions.
-  s3cmd put --follow-symlinks "$build_dir/deb_repo/keybase-latest-amd64.deb" "s3://$BUCKET_NAME/keybase_amd64.deb"
-  s3cmd put --follow-symlinks "$build_dir/deb_repo/keybase-latest-i386.deb" "s3://$BUCKET_NAME/keybase_i386.deb"
-  s3cmd put --follow-symlinks "$build_dir/rpm_repo/keybase-latest-x86_64.rpm" "s3://$BUCKET_NAME/keybase_amd64.rpm"
-  s3cmd put --follow-symlinks "$build_dir/rpm_repo/keybase-latest-i386.rpm" "s3://$BUCKET_NAME/keybase_i386.rpm"
-
-  json_tmp=`mktemp`
-  echo "Writing version into JSON to $json_tmp"
-
-  "$release_bin" update-json --version="$version" > "$json_tmp"
-
-  s3cmd put --mime-type application/json "$json_tmp" "s3://$BUCKET_NAME/update-linux-prod.json"
-
-  # Generate and push the index.html file. S3 pushes in this script can be
-  # flakey, and on the Linux side of things all this does is update our
-  # internal pages, so we suppress errors here.
-  GOPATH="$release_gopath" PLATFORM="linux" "$here/../prerelease/s3_index.sh" || \
-    echo "ERROR in s3_index.sh. Internal pages might not be updated. Build continuing..."
-
-  "$here/arch/update_aur_packages.sh" "$build_dir"
-}
-
-if [ "$mode" = "prerelease" ] ; then
-  release_prerelease
-else
-  release_serverops
-fi
+echo Doing a prerelease push to S3...
+
+# Parse the shared .s3cfg file and export the keys as environment variables.
+# (Our s3cmd commands would be happy to read that file directly if we put it
+# in /root, but the s3_index.sh script ends up running Go code that depends
+# on the variables.)
+export AWS_ACCESS_KEY="$(grep access_key ~/.s3cfg | awk '{print $3}')"
+export AWS_SECRET_KEY="$(grep secret_key ~/.s3cfg | awk '{print $3}')"
+
+# Upload both repos to S3. We don't use the --delete-removed flag, because
+# clients with cached repo data might try to fetch older binaries, and we
+# prefer to let them do that rather than giving them nasty errors. They
+# should eventually get the latest stuff when they run the equivalent of
+# `apt-get update`. See https://github.com/keybase/keybase-issues/issues/2816.
+echo Syncing the deb repo...
+s3cmd sync --add-header="Cache-Control:max-age=60" "$build_dir/deb_repo/repo/" "s3://$BUCKET_NAME/deb/"
+echo Syncing the rpm repo...
+s3cmd sync --add-header="Cache-Control:max-age=60" "$build_dir/rpm_repo/repo/" "s3://$BUCKET_NAME/rpm/"
```

Comment:
The flags are deleted here. Everything else in this huge chunk is an indentation change.